AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -245,6 +267,8 @@ - (void)reloadDataWithAnimationOptions:(ASDataControllerAnimationOptions)animati
       NSMutableArray *updatedIndexPaths = [NSMutableArray array];
       [self _populateFromEntireDataSourceWithMutableNodes:updatedNodes mutableIndexPaths:updatedIndexPaths];
       
+      [self _layoutNodesWithMainThreadAffinity:updatedNodes atIndexPaths:updatedIndexPaths];
```

Comment:
We can finish syncing up on this in the new #736 but in short, I think the problem here is naming. The methods are named `_layoutNodes:` but what we're really doing is measuring them, so that we will be able to return the correct heightForRowAtIndexPath: or sizeForItemAtIndexPath:. Before this PR, we always measured nodes in the background before we reported them to the UIKit class. But if the node's view is already loaded when data controller gets told about it then it's unsafe to remeasure it in the background so we do it before we leave the main thread.